### DISCUSSION TITLE
Differences between Kaoto and Karavan?

### DISCUSSION THREAD

**Discussion by vincentytran** [KaotoIO/kaoto#1434] [General]
To me they look so similar and try to do similar things. Is there a history between Kaoto and Karavan or just by chance?
Do they work against the same .yaml file so you can use both at the same time?

**Comment by lhein** [comment]
Both projects started around the same time independently and both evolved over time. There are also other Camel editors like the Fuse Eclipse tooling as well as another VS Code extension named Camel Designer. Probably there are even more we don't have on the radar. 

While the eclipse based editor was based on XML DSL the newer ones use Yaml DSL. 

I hope that helps.

Lars

**Comment by skin27** [comment]
Kaoto and Karavan are both visual editors to edit routes, route configuration, beans, rest, and error handlers. Basically visualizing everything that is the Camel catalog. They both use the YAML DSL.

1. Visual Kaoto uses bigger icons and then prepend/append new nodes, whereas Karavan uses smaller icons and use plus sign to add new nodes.
2. Kaoto has also modules that abstract away from routes, using Kamelets and Pipes.
